<a href="https://colab.research.google.com/github/athallfs/1-04-2026/blob/main/PraktikumEDA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

-----------------------------------

**LATIHAN 1**

In [ ]:
import numpy as np

harga = np.array([5000, 7000, 3000, 12000, 4500])
print('Rata-rata harga:', harga.mean())
print('Harga tertinggi:', harga.max())
print('Harga setelah diskon 10%:', harga * 0.9)

Rata-rata harga: 6300.0
Harga tertinggi: 12000
Harga setelah diskon 10%: [ 4500.  6300.  2700. 10800.  4050.]


Jika menggunakan NumPy array, harga * 0.9 akan mengalikan setiap elemen dengan 0,9 sehingga semua harga otomatis mendapat diskon 10%. Sedangkan [5000,7000,3000] * 0.9 pada Python list akan menghasilkan TypeError karena list tidak mendukung perkalian dengan bilangan desimal. Pada list, operator * digunakan untuk mengulang isi list dengan bilangan bulat. NumPy array lebih cocok untuk operasi perhitungan pada banyak data karena mendukung operasi secara langsung pada setiap elemen.

**LATIHAN 2**

In [ ]:
import pandas as pd
data_kantin = {
    'menu': ['Nasi Goreng', 'Es Teh', 'Mie Ayam', 'Es Teh', None],
    'harga': [12000, 4000, 10000, 4000, 8000],
    'terjual': [23, 40, None, 35, 18]
}
df = pd.DataFrame(data_kantin)
print(df)

          menu  harga  terjual
0  Nasi Goreng  12000     23.0
1       Es Teh   4000     40.0
2     Mie Ayam  10000      NaN
3       Es Teh   4000     35.0
4         None   8000     18.0


Kolom yang memiliki data kosong adalah menu dan terjual. Data kosong pada kolom menu ditampilkan sebagai None, sedangkan pada kolom terjual ditampilkan sebagai NaN. Jika data langsung dianalisis tanpa dibersihkan, hasil perhitungan seperti total, rata-rata, atau jumlah penjualan dapat menjadi tidak akurat karena ada data yang hilang. Oleh karena itu, data kosong perlu diperiksa dan ditangani terlebih dahulu.

**LATIHAN 3**


In [ ]:
df = pd.read_csv('data_kantin.csv')

print(df.head()) # 5 baris pertama
print(df.info()) # tipe data & jumlah non-null tiap kolom
print(df.describe()) # statistik ringkas kolom numerik
print(df.shape) # jumlah (baris, kolom)

      tanggal         menu kategori  harga  terjual
0  2026-08-20   Teh Hangat  Minuman   3000     11.0
1  2026-08-31  Nasi Goreng  Makanan  12000     29.0
2  2026-08-21  Nasi Goreng  Makanan  12000     28.0
3  2026-08-28   Teh Hangat  Minuman   3000     12.0
4  2026-08-21     Es Jeruk  Minuman   5000     19.0
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    object 
 1   menu      107 non-null    object 
 2   kategori  110 non-null    object 
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 4.4+ KB
None
              harga     terjual
count    110.000000  104.000000
mean    6336.363636   22.923077
std     3520.001232   10.831178
min     2000.000000    8.000000
25%     3000.000000   13.000000
50%     5000.000000   20.000000
75%    10000.000000 

Kolom yang jumlah non-null-nya lebih sedikit dari jumlah baris total adalah menu dan terjual. Kolom menu memiliki 107 data terisi dari 110 baris, sehingga terdapat 3 data kosong. Kolom terjual memiliki 104 data terisi, sehingga terdapat 6 data kosong. Artinya, kedua kolom tersebut memiliki missing value (data yang hilang) yang perlu dibersihkan atau ditangani sebelum dilakukan analisis lebih lanjut.

**LATIHAN 4**

In [4]:
print(df.isnull().sum()) # jumlah data kosong tiap kolom
df['terjual'] = df['terjual'].fillna(0) # isi kekosongan dengan 0
df = df.dropna(subset=['menu']) # hapus baris jika kolom menu kosong


tanggal     0
menu        0
kategori    0
harga       0
terjual     0
dtype: int64


Kolom terjual diisi dengan 0 karena data yang kosong masih bisa dianggap sebagai tidak ada penjualan yang tercatat, sehingga data tetap dapat digunakan untuk perhitungan. Sedangkan baris dengan menu kosong dihapus karena nama menu merupakan informasi penting untuk mengetahui barang apa yang dijual. Jika nama menu tidak diketahui, data tersebut sulit digunakan dan dapat membuat analisis menjadi tidak jelas.

**LATIHAN 5**

In [5]:
print(df.duplicated().sum()) # jumlah baris duplikat
df = df.drop_duplicates()
df['harga'] = df['harga'].astype(int) # memastikan tipe data harga adalah integer
print(df.dtypes)

5
tanggal      object
menu         object
kategori     object
harga         int64
terjual     float64
dtype: object


Setelah drop_duplicates(), jumlah baris dataset menjadi lebih sedikit atau tetap sama dibanding sebelumnya, tergantung apakah terdapat baris yang duplikat. Baris yang sama persis akan dihapus sehingga tidak dihitung berulang. Memastikan tipe data (dtypes) benar penting agar proses analisis dan perhitungan berjalan dengan tepat. Misalnya, kolom harga harus bertipe integer agar dapat digunakan untuk operasi matematika seperti menghitung total atau rata-rata harga tanpa masalah.

**LATIHAN 6**

In [6]:
laris = df[df['terjual'] > 20] # filtering
urut = df.sort_values(by='terjual', ascending=False) # sorting
df['total_pendapatan'] = df['harga'] * df['terjual'] # kolom turunan
ringkasan = df.groupby('menu')['total_pendapatan'].sum() # agregasi
print(ringkasan)

menu
Es Jeruk       1460000.0
Es Teh         2144000.0
Kerupuk         972000.0
Mie Ayam       2410000.0
Nasi Goreng    3924000.0
Roti Bakar     1288000.0
Teh Hangat      459000.0
Name: total_pendapatan, dtype: float64


Menu yang menghasilkan total pendapatan tertinggi adalah Nasi Goreng, dengan total pendapatan sebesar Rp3.924.000.

Informasi ini dapat membantu kantin sekolah menentukan menu yang perlu diprioritaskan, menjaga ketersediaan stok bahan baku, dan mengatur strategi penjualan agar pendapatan kantin meningkat.

Sebagai langkah berikutnya, kantin juga bisa membandingkan jumlah porsi terjual untuk mengetahui apakah pendapatan tinggi disebabkan oleh banyaknya penjualan atau harga per porsi.